logbus/solvermon/vertexmon.go

```diff
@@ -37,111 +37,147 @@ type vertexMonitor struct {
-		errString = fmt.Sprintf(""+
+		return fmt.Sprintf(
 			"      The%s command\n"+
-			"          %s\n"+
-			"      failed: %s",
-			internalStr, indentOp, m[2])
-		vm.isFatalError = true
-		vm.fatalErrorType = logstream.FailureType_FAILURE_TYPE_FILE_NOT_FOUND
-	case errString == "no active sessions":
-		vm.isCanceled = true
-		errString = "WARN: Canceled"
-	case strings.Contains(errString, errutil.EarthlyGitStdErrMagicString):
+				"          %s\n"+
+				"      failed: %s", internal, operation, m[2])
+	case logstream.FailureType_FAILURE_TYPE_GIT:
 		gitStdErr, shorterErr, ok := errutil.ExtractEarthlyGitStdErr(errString)
 		if ok {
-			errString = fmt.Sprintf(
-				"The%s command '%s' failed: %s\n\n%s", internalStr, vm.operation, shorterErr, gitStdErr)
-		} else {
-			errString = fmt.Sprintf(
-				"The%s command '%s' failed: %s", internalStr, vm.operation, errString)
+			return fmt.Sprintf(
+				"The%s command\n"+
+					"          %s\n"+
+					"failed: %s\n\n%s", internal, operation, shorterErr, gitStdErr)
 		}
-		vm.isFatalError = true
+		return fmt.Sprintf(
+			"The%s command\n"+
+				"          %s\n"+
+				"failed: %s", internal, operation, errString)
 	default:
-		errString = fmt.Sprintf(
-			"The%s command '%s' failed: %s", internalStr, vm.operation, errString)
+		return fmt.Sprintf(
+			"The%s command\n"+
+				"          %s\n"+
+				"failed: %s", internal, operation, errString)
 	}
+}
+
+func FormatError(operation string, errString string) string {
+	exitCode := getExitCode(errString)
+	fatalErrorType, _ := determineFatalErrorType(errString, exitCode)
+	return formatErrorMessage(errString, operation, "", fatalErrorType, exitCode)
+}
+
+func (vm *vertexMonitor) parseError() {
+	errString := vm.vertex.Error
+
+	indentOp := strings.Join(strings.Split(vm.operation, "\n"), "\n          ")
+	internalStr := ""
+	if vm.meta.Internal {
+		internalStr = " internal"
+	}
+	errString = fmt.Sprintf("%s%s", internalStr, errString)
+
+	exitCode := getExitCode(errString)
+	fatalErrorType, isFatalError := determineFatalErrorType(errString, exitCode)
+	formattedError := formatErrorMessage(errString, indentOp, internalStr, fatalErrorType, exitCode)
+
+	// Add Error location
 	slString := ""
 	if vm.meta.SourceLocation != nil {
 		slString = fmt.Sprintf(
 			" %s:%d:%d",
 			vm.meta.SourceLocation.File, vm.meta.SourceLocation.StartLine,
 			vm.meta.SourceLocation.StartColumn)
 	}
-	if vm.isFatalError {
-		vm.errorStr = fmt.Sprintf("ERROR%s\n%s", slString, errString)
+
+	// Set the error string and flags on the vertexMonitor
+	if isFatalError {
+		vm.errorStr = fmt.Sprintf("ERROR%s\n%s", slString, formattedError)
 	} else {
-		vm.errorStr = fmt.Sprintf("WARN%s: %s", slString, errString)
+		vm.errorStr = fmt.Sprintf("WARN%s: %s", slString, formattedError)
 	}
+	vm.isFatalError = isFatalError
+	vm.fatalErrorType = fatalErrorType
+}
+
+func (vm *vertexMonitor) Write(dt []byte, ts time.Time, stream int) (int, error) {
```

Comment:
my guess is this function wasn't changed, just moved?

Reply:
Yeah, my bad. Fixed.